pkg/pquerier/frames.go

```diff
@@ -355,14 +355,31 @@ func (d *dataFrame) TimeSeries(i int) (utils.Series, error) {
 // Creates Frames.columns out of tsdb columns.
 // First do all the concrete columns and then the virtual who are dependant on the concrete.
 func (d *dataFrame) finishAllColumns() error {
-
 	// Marking as deleted every index (row) that has no data.
-	for i, hasData := range d.nonEmptyRowsIndicators {
+	// Also, adding "blank" rows when needed to align all columns to the same time.
+	// Iterating backwards to not miss any deleted cell.
+	for i := len(d.nonEmptyRowsIndicators) - 1; i >= 0; i-- {
+		hasData := d.nonEmptyRowsIndicators[i]
 		if !hasData {
 			for _, col := range d.columns {
 				_ = col.Delete(i)
 			}
 			_ = d.index.Delete(i)
+		} else {
+			for _, col := range d.columns {
+				switch currCol := col.(type) {
+				case *dataColumn:
```

Comment:
I see both cases are equaly treated, why not do "case *dataColumn, *ConcreteColumn :" ?